app/services/ipam/base.rb

```diff
@@ -41,6 +41,18 @@ def serializable_hash(options = nil)
 
     private
 
+    def block_ip_cache_key(ip)
+      "blocked_ip_cache_#{subnet.id}/#{ip}"
+    end
+
+    def block_ip(ip)
+      Rails.cache.write(block_ip_cache_key(ip), 'blocked', expires_in: 30.minutes)
+    end
+
+    def is_ip_blocked?(ip)
+      Rails.cache.read(block_ip_cache_key(ip)) == 'blocked'
```

Comment:
Isn't this sufficient?
```suggestion
      Rails.cache.exists?(block_ip_cache_key(ip))
```

In some cache backends it may be cheaper to test for existence than reading the value.

Reply:
you are right. Thank you!